cg/models/flow_cell/flow_cell.py

```diff
@@ -238,6 +249,44 @@ def is_flow_cell_ready(self) -> bool:
         LOG.info(f"Flow cell {self.id} is ready for downstream processing")
         return True
 
+    def get_run_completion_status(self) -> Path | None:
+        """Return the run completion status path."""
+        flow_cells_dir: Path = self.get_flow_cell_run_dir()
+        file_path = Path(flow_cells_dir, DemultiplexingDirsAndFiles.RUN_COMPLETION_STATUS)
+        if file_path.exists():
+            return file_path
+        return None
+
+    @property
+    def sequencing_started_at(self) -> datetime.datetime | None:
+        parser = ParseRunCompletionStatusService()
+        file_path: Path = self.get_run_completion_status()
+        return parser.get_start_time(file_path) if file_path else None
+
+    @property
+    def sequenced_completed_at(self) -> datetime.datetime | None:
+        parser = ParseRunCompletionStatusService()
+        file_path: Path = self.get_run_completion_status()
+        return parser.get_end_time(file_path) if file_path else None
+
+    @property
+    def demultiplexing_started_at(self) -> datetime:
+        """Get the demultiplexing started time stamp from the flow cell run dir."""
+        flow_cell_dir = self.get_flow_cell_run_dir()
+        demux_started_path = Path(flow_cell_dir, DemultiplexingDirsAndFiles.DEMUX_STARTED)
```

Comment:
this variable is already a property of this class (`demultiplexing_started_path`). It uses `self.path`, maybe modify it to use the `flow_cell_run_directory` that you defined above